### ISSUE TITLE
External reference to PDF #subsection

### ISSUE THREAD

**Issue by sgkoishi** [typst/typst#1352]
### Description

Some PDFs support linking to a subsection from outside the file (e.g. in URL).
Example: http://tug.ctan.org/macros/latex/contrib/beamer/doc/beameruserguide.pdf#subsection.14.1

Typst doesn't generate such metadata and can't have a link to a specific section in PDF. 
Might be related to #133

### Use Case

Users can link to specific sections in the pdf like the example.

_There is a Firefox-only link that you can copy from the outline, and it looks like `#[{"num":index,"gen":0},{"name":"XYZ"},position,position,0]` if no `#subsection` metadata is provided_

**Comment by ondohotola** [comment]
See

[latex pdfsync](https://ctan.org/tex-archive/macros/latex/contrib/pdfsync?lang=en) and [skim TeX-PDF Synchronization](https://sourceforge.net/p/skim-app/wiki/TeX_and_PDF_Synchronization/)

**Comment by reknih** [comment]
This relies on "Named Destinations" in the document catalog of which we do not write any at the moment. Refer to PDF 1.7 spec, 12.3.2.3 and https://github.com/mozilla/pdf.js/issues/5258.

An open question is what name the destination would be given since they should best be URL safe without escapes. Should this be user-configurable?

**Comment by Heinenen** [comment]
> Do other elements than headings create named destinations?
I would say yes, e.g. citations, figures

> An open question is what name the destination would be given since they should best be URL safe without escapes. 
For built-in it would be nice to have escape-free URLs which seems achievable for most elements like headings. An open problem may be custom numbering functions which require escapes.

> Should this be user-configurable?
I don't think it would be the most crucial part to configure but it does sound like a nice thing to configure for citations (the default behavior for LaTeX seems to be the citation key, but one may like something else better)
A big benefit of making it configurable (from the start) would be that we wouldn't have to decide on what elements we wanted it from the get-go, the user could decide. We also don't have to decide on a good naming scheme to avoid escapes, as that would be on the user too. This would still be open to discussion without being a breaking change later on.

TLDR: If we make it configurable we could afford to postpone dealing with the other questions, and everything is better than the current "my.pdf#%5B%7B%22num%22%3A6%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C70.837906%2C620.16907%2C0%5D" :D
